### PR TITLE
2941 timeout for geocodings

### PR DESCRIPTION
**Pull request by rafatower**
@Kartones can you please review?

Things done here:
- Change DB timeout to 100 minutes. Why? because max processing time since we got metrics is that for a failing geocoding. So most likely that time should be 1 order of magnitude smaller, at most.
- Set a processing time timeout to 200 minutes, double of max execution time for a failing geocoding.
- Use Ruby's stdlib http://ruby-doc.org/stdlib-1.9.3/libdoc/timeout/rdoc/Timeout.html to implement the timeout
- Add a test for it (not really necessary IMO since it relies on the library and I also tested it manually but a convenience for the future).

### CODE REVIEW

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3583/))

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3588/))

File: app/models/geocoding.rb [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
+require 'timeout'
 require_relative '../../services/table-geocoder/lib/table_geocoder'
 require_relative '../../services/table-geocoder/lib/internal_geocoder.rb'
 require_relative '../../lib/cartodb/metrics'
 
 class Geocoding < Sequel::Model
 
-  DB_TIMEOUT      = 3600000*24*2  # Way generous, 2 days is a lot
+  DB_TIMEOUT_MS              = 100.minutes.to_i * 1000
+  PROCESSING_TIMEOUT_SECONDS = 200.minutes.to_i
```
Comment (Kartones) [inline app/models/geocoding.rb]: if is double of DB_TIMEOUT_MS, will it work this?
```
PROCESSING_TIMEOUT_SECONDS = DB_TIMEOUT_MS / 1000
```

File: app/models/geocoding.rb [hunk 2]
```diff
@@ -20,6 +22,8 @@ class Geocoding < Sequel::Model
   attr_reader :table_geocoder
   attr_reader :started_at, :finished_at
 
+  class ProcessingTimeoutException < StandardError; end
```
Comment (Kartones) [inline app/models/geocoding.rb]: I prefer adding Exception classes after main file class but personal taste

**Comment by Kartones** [comment]
:+1:  and nice test for checking timeouts!

File: app/models/geocoding.rb (hunk 1 above)
Comment (rafatower) [inline app/models/geocoding.rb]: `PROCESSING_TIMEOUT_SECONDS = DB_TIMEOUT_MS / 1000 * 2`

But ok, as you see is more clear

File: app/models/geocoding.rb (hunk 2 above)
Comment (rafatower) [inline app/models/geocoding.rb]: ok

File: app/models/geocoding.rb (hunk 1 above)
Comment (rafatower) [inline app/models/geocoding.rb]: Mmm I prefer to keep it that way, cause they shouldn't be directly related.

I just chose some values that made sense in my mind with regards to existing metrics, but do not want someone stumbling with that code and thinking these timeouts must necesarily be related to one another by a given factor.

Comment (Kartones) [inline app/models/geocoding.rb]: Ok, fair enough

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3596/))

**Pull request merged**